pkg/internal/cryptoutil/utils.go

```diff
@@ -117,6 +122,67 @@ func lengthPrefix(array []byte) []byte {
 	return arrInfo
 }
 
+// Curve25519KeySize number of bytes in a Curve25519 public or private key
+const Curve25519KeySize = 32
+
+// NonceSize size of a nonce used by Box encryption (Xchacha20Poly1305)
+const NonceSize = 24
+
+// PublicEd25519toCurve25519 takes an Ed25519 public key and provides the corresponding Curve25519 public key
+//  This function wraps PublicKeyToCurve25519 from Adam Langley's ed25519 repo: https://github.com/agl/ed25519
+func PublicEd25519toCurve25519(pub []byte) ([]byte, error) {
```

Comment:
this function can stay in utils.go as it will be used by jwe encryption too (when we move envelope keys encryption to signature keys)